File: Data Structures & Algorithms/search-2d-matrix/submission-7.js
class Solution {
    /**
     * @param {number[][]} matrix
     * @param {number} target
     * @return {boolean}
     */
    searchMatrix(matrix, target) { 
        let cols = matrix[0].length 
        let row = matrix.length 

        let l = 0 
        let r = cols * row - 1 

        while(l <= r) {
            let m = Math.floor(( r + l ) / 2) 
            let midVal = matrix[Math.floor(m / cols)][m % cols] 

            if (midVal === target) {
                return true
            } else if (midVal < target) {
                l = m + 1
            } else {
                r = m - 1
            }
        } 

        return false
    }
}

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-7.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        rows = len(matrix) 
        cols = len(matrix[0]) 

        x = [1] * cols
        y = [1] * rows 

        for r in range(rows): 
            for c in range(cols): 
                if matrix[r][c] == 0: 
                    y[r] = 0 
                    x[c] = 0 

        for r in range(rows): 
            for c in range(cols): 
                if x[c] == 0 or y[r] == 0: 
                    matrix[r][c] = 0